### PR TITLE
Define expression that `Amount` class takes in.

### PR DESCRIPTION
**Pull request by eugeneyl**
Changed `Amount` class to ensure that each amount:
- Starts with a `$` sign.
- Can take in either 2 decimal place or no decimals at all.

Updated test cases based on above-mentioned changes.

### CODE REVIEW

File: src/test/java/seedu/finance/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
     public static final String CATEGORY_DESC_HUSBAND = " " + PREFIX_CATEGORY + VALID_CATEGORY_HUSBAND;
 
     public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "James&"; // '&' not allowed in names
-    public static final String INVALID_AMOUNT_DESC = " " + PREFIX_AMOUNT + "*2"; // "*" not allowed in amounts
+    public static final String INVALID_AMOUNT_DESC = " " + PREFIX_AMOUNT + "42"; // '*' not allowed in amount
```
Comment (kjiaxuan) [inline src/test/java/seedu/finance/logic/commands/CommandTestUtil.java]: I think there may be a missing * in the string

Comment (eugeneyl) [reply src/test/java/seedu/finance/logic/commands/CommandTestUtil.java]: Corrected! Thanks for highlighting!

File: src/test/data/JsonSerializableFinanceTrackerTest/typicalRecordsFinanceTracker.json [hunk 2]
```diff
@@ -2,43 +2,43 @@
...
     "name" : "Alice Pauline",
-    "amount" : "120",
+    "amount" : "$120.30",
     "date" : "12/02/2017",
     "description": "",
     "tagged" : [ "friends" ]
   }, {
     "name" : "Benson Meier",
-    "amount" : "119",
+    "amount" : "$119",
```
Comment (geezlouisee) [inline src/test/data/JsonSerializableFinanceTrackerTest/typicalRecordsFinanceTracker.json:11]: * Value here not stated in 2dp, unlike in `src/test/data/JsonSerializableFinanceTrackerTest/typicalRecordsFinanceTracker.json` where prices listed as absolute values are rounded off to 2dp - should it be standardised to 2dp as well? (hence becomes `$119.00` , similarly for the other test cases in this file too)

File: src/main/java/seedu/finance/logic/commands/SpendCommand.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
             + "[" + PREFIX_CATEGORY + "CATEGORY]...\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "John Doe "
-            + PREFIX_AMOUNT + "123 "
+            + PREFIX_AMOUNT + "$123.23 "
```
Comment (Jackimaru96) [inline src/main/java/seedu/finance/logic/commands/SpendCommand.java:31]: Should not have `$` as it is not intuitive for user to use `PREFIX_AMOUNT` (`$/`).

**Pull request merged**